src/App.jsx

```diff
@@ -1,11 +1,127 @@
+
+            console.log(`Country: ${data[0].name.common}, Capital: ${data[0].capital[0]}`);
+
+            if (!data || data.length === 0) {
+            setError(`${name} bestaat niet. Probeer het opnieuw.`);
+            setCountryData(null);
+            } else {
+                setCountryData(data[0]);
+                setError(''); // Verwijder foutmelding bij succesvolle zoekopdracht
+            }
+        } catch (error) {
+            console.error('Error fetching country:', error);
+            setError('Er is een fout opgetreden. Probeer het opnieuw.');
+        }
+    };
+
+
+    const handleSearch = async () => {
+        if (searchTerm) {
+            await fetchCountry(searchTerm);
+            setSearchTerm(''); //Leeg het zoekveld na zoekopdracht
+        }
+    };
+
+    const handleKeyDown = (event) => {
+        if (event.key === 'Enter') {
+            handleSearch();
+        }
+    };
+
     return (
-        <>
-            Maak je applicatie hier!
-        </>
+        <div className="App">
+            <h1>Country Information</h1>
+            <img src={worldMap} alt="WorldMap" style={{ width: '100%', maxHeight: '500px', objectFit: 'cover' }} />
+
+            {/*} //Functionaliteit Opdracht 1 - weergeven van alle landen
+            {!isFetched && (
+                <button onClick={fetchCountries}>Fetch Countries</button>)}
+
+            <div className="countries-container">
+                {countries.map((country) => (
+                    Voor het identificeren van landen in deze .map functie wordt de unieke cca3 code die elk land heeft gebruikt. Dit zorgt ervoor dat React efficienter met de lijst omgaat.
+                    <div key={country.cca3}>
+                        <img src={country.flags.svg} alt={`Flag of ${country.name.common}`} width="100" />
+                        <p style={{ color: getColorByRegion(country.region) }}>{country.name.common}</p>
+                        <p>Has a population of {country.population.toLocaleString()} people</p>
+                    </div>
+                ))}
+            </div>
+        */}
+
+            {/* Functionaliteit Opdracht 2 */
+            }
+            <div className="search-container">
+                <input
+                    type="text"
+                    value={searchTerm}
+                    onChange={(e) => setSearchTerm(e.target.value)}
+                    onKeyDown={handleKeyDown}
+                    placeholder="Search for a country..."
+                    />
+                <button onClick={handleSearch}>Search</button>
+            </div>
+
+            {error && <p className="error">{error}</p>}
+
+            {countryData && (
+                <div className="country-info">
+                    <img src={countryData.flags.svg} alt={`Flag of ${countryData.name.common}`} width="100" />
+                    <h2 style={{ color: getColorByRegion(countryData.region) }}>{countryData.name.common}</h2>
```

Comment:
Hier gebruik je opeens inline styling. Ik snap waarom, want hierdoor hoef je minder regels css te schrijven. Consistentie is echter belangrijker wat betreft best practice.